File: sea_app/lib/models/result_understanding.dart
class ResultUnderstanding {
  final String id;
  final String type;
  final String? fkIdSummaryDiscussion;

  ResultUnderstanding({required this.id, required this.type, this.fkIdSummaryDiscussion});

  static ResultUnderstanding fromJson(Map<String, dynamic> j) {
    return ResultUnderstanding(
      id: (j['id_resultunderstanding'] ?? j['id'] ?? '').toString(),
      type: (j['type'] ?? '').toString(),
      fkIdSummaryDiscussion: (j['fk_id_summarydiscussion'] ?? j['fk_id_summarydiscussion'])?.toString(),
    );
  }
}

File: backend/app/Models/ResultUnderstanding.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ResultUnderstanding extends Model
{
    use HasFactory;

    protected $table = 'result_understandings';
    protected $primaryKey = 'id_resultunderstanding';
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'type',
        'fk_id_summarydiscussion',
    ];
}
